### PR TITLE
[APPTS-9277] Inherit Env props from process.env vars

### PR DESCRIPTION
**Pull request by jareddlc**
Backporting changes needed for onprem:

https://jira.appcelerator.org/browse/APPTS-9277

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -140,6 +140,16 @@ DEFAULTS.PLATFORM_AXWAY = {
 	secureCookies: true
 };
 
+var PLATFORM_MAP = {
+	baseurl: 'APPC_DASHBOARD_URL',
+	registryurl: 'APPC_REGISTRY_URL',
```
Comment (pkclark) [inline lib/index.js]: @jareddlc is sec server's URL needed in this mapping?

Comment (jareddlc) [inline lib/index.js]: good idea, we should probably add it

**Comment by jareddlc** [comment]
can someone confirm the tests pass. They fail a login test on my system

**Comment by pkclark** [comment]
@jareddlc using the dev conf from the README?

File: lib/index.js [hunk 2]
```diff
@@ -140,6 +140,17 @@ DEFAULTS.PLATFORM_AXWAY = {
...
```
Comment (pkclark) [inline lib/index.js]: @jareddlc this supposed to be APPC_REGISTRY_SERVER?

Comment (jareddlc) [inline lib/index.js]: i was basing it off this pr: https://github.com/appcelerator/appc-platform-sdk/pull/54

File: test/appc_platform_sdk_tests.js [hunk 3]
```diff
@@ -80,6 +80,29 @@ describe('appc-platform-AppC', function () {
 			should(AppC.cacheurl).be.equal('http://cache.com');
 			should(AppC.pubsuburl).be.equal('http://pubsub.com');
 		});
+
+		it('should be able to use environment variables overrides', function () {
```
Comment (feons) [inline test/appc_platform_sdk_tests.js]: Probably it's good to reset the env var settings at the end of the test

Comment (jareddlc) [inline test/appc_platform_sdk_tests.js]: fixed tests

**Comment by feons** [comment]
Tests passed locally for me.

File: lib/index.js (hunk 2 above)
Comment (pkclark) [inline lib/index.js]: @feons appc-cli is reading this as APPC_REGISTRY_SERVER, isn't it?

Comment (pkclark) [inline lib/index.js]: #54 may be wrong in that case.

Comment (feons) [inline lib/index.js]: yup it should be APPC_REGISTRY_SERVER.

Comment (jareddlc) [inline lib/index.js]: should I revert to APPC_REGISTRY_SERVER and APPC_SECURITY_SERVER ?

Comment (pkclark) [inline lib/index.js]: @jareddlc yes please.  A PR against master would be necessary too to correct this oversight.

Comment (feons) [inline lib/index.js]: submitting a pr against master. :(

**Comment by jareddlc** [comment]
accidentally included previous commmit

**Comment by feons** [comment]
try `git reflog`

**Comment by feons** [comment]
*LGTM.*

**Comment by pkclark** [comment]
This should perhaps be targeting a "1.X" branch.  Going forward, any introduced branches should align with this project's versions, not consumers'.

**Pull request merged**